### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by AmineDerbal**
In this pull request ♻️ 

- Solve the N+1 problem when fetching in this URL (users/:id/posts) 👍 
- Setup Capybara to write integration tests 👍 
- write all required tests 👍 
- Use this command for tests **rails test:system** 👍 
- loom video link for test demonstration https://www.loom.com/share/9bc26fbee3ea40a89db5d151715a00c9?sid=0394aa5f-be4a-4884-8fde-778c7aa88a9e 👍 

### Screenshot app's log before fix 
posts-before-include
### Screenshot app's log after fix 
posts-after-include

### CODE REVIEW

File: test/system/posts_test.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    # Expectation: I can see how many comments it has
+    assert_text "Comments: #{post.comments.count}"
+
+    # Expectation: I can see how many likes it has
+    assert_text "Likes: #{post.likes.count}"
+
+    # Expectation: I can see the username of each commentor.
+    post.comments.each do |comment|
+      assert_text comment.author.name
```
Comment (DuaneDave) [inline test/system/posts_test.rb]: Good job writing the integration test 👍, however, you forgot to write for a scenario that asserts you can see the post body as stated in the project [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20the%20post%20body.). Kindly add the test in order to satisfy the project requirements.

File: test/system/posts_test.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      assert_text comment.text
+    end
+
+    # Expectation: I can see the comment each commentor left.
```
Comment (DuaneDave) [inline test/system/posts_test.rb]: Consider writing the test that asserts that you can see the comment each commentor left in order to satisfy the project requirement

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @AmineDerbal and @FaranoshAmini 👋 , 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Correct use of git-flow
✔️ You did a good job writing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: test/system/posts_test.rb (hunk 2 above)
Comment (AmineDerbal) [reply test/system/posts_test.rb]: that was done all comments of a post are tested within this code @DuaneDave 
   ```
  post.comments.each do |comment|
      # Expectation: I can see the username of each commentor.
      assert_text comment.author.name
      # Expectation: I can see the comment each commentor left.
      assert_text comment.text
    end
```

File: test/system/posts_test.rb (hunk 1 above)
Comment (AmineDerbal) [reply test/system/posts_test.rb]: Will do!!

**Comment by AmineDerbal** [comment]
-  a scenario that asserts you can see the post body in the Post show page 👍 
-   the test that asserts that you can see the comment each commentator left in the Post show page is 
from line 72 to 77 👍 
post-comment-test

- New loom video link for test demonstration https://www.loom.com/share/2f0c923392e7460fa7da7db332d61697?sid=ba173cb7-5687-4934-a336-a442ea275936 👍

**Review by jerryowusu — APPROVED** [review]
Hi @AmineDerbal 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Used gitflow
:heavy_check_mark: Tests passing

CookedDelectableDolphin-size_restricted


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**